dashboard/app/models/channel_token.rb

```diff
@@ -17,4 +17,46 @@
 class ChannelToken < ActiveRecord::Base
   belongs_to :user
   belongs_to :level
+
+  def self.create_channel_token(level, user, ip, data = {})
+    # The channel should be associated with the template level, if present.
+    # Otherwise the current level.
+    host_level = level.project_template_level || level
```

Comment:
Nit: could de-dupe with [L38-40](https://github.com/code-dot-org/code-dot-org/pull/9177/files#diff-50313fe978d94f948471a7a4b51dd7a6R40) as a private method:
```ruby
# The channel should be associated with the template level, if present.
# Otherwise the current level.
private_class_method def self.host_level(level)
  level.project_template_level || level
end
```
Or as a method on `level`:
```ruby
# (in level.rb)
def host_level
  project_template_level || self
end
```